### DISCUSSION TITLE
Heading: Template Entity

### DISCUSSION THREAD

**Discussion by Nixon506E** [piitaya/lovelace-mushroom#1548] [Ideas]
It would be great to implement the a template entity for the heading card similar to the way the template badge was added. 

I was hoping to have different icon color states but currently am unable to and the muchroom template badges provide so much functionality that would be great here too.

**Comment by BBE-FR** [comment]
Same request has been made in the frontend to extend the capability of the header card:
https://github.com/home-assistant/frontend/discussions/22243

I am afraid that there maybe more chance to have this capability here in mushroom for an advanced "mushroom-header-card" offering the typical mushroom templating capabilities, tap, hold and double tap actions...

Hope to see an implementation of it.

Nevertheless, it is already possible to customise a little bit the icons for the entities displayed, with themes using state-color for the displayed entities type:
https://www.home-assistant.io/integrations/frontend#state-color
```
  #door
  state-binary_sensor-door-off-color: green
  state-binary_sensor-door-on-color: red
  #motion
  state-binary_sensor-motion-off-color: green
  state-binary_sensor-motion-on-color: red
```

But this does not allow templating options (or you have to put the template into a dedicated helper)
And this does not affect the "main" icon and title, only the entities one (on the right).
![image](https://github.com/user-attachments/assets/c672b616-1726-4b54-bddc-9d0e2f75fcbe)

**Reply by flavioxavier** [reply]
I also support implementing this, a template for the title name and icon is honestly a must have to make use of the heading card, hope mushroom implements it.

Adding template to badges would also be great.

**Comment by BBE-FR** [comment]
I have created a feature request for this instead of a discussion:
https://github.com/piitaya/lovelace-mushroom/issues/1556
